spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/DenseTensor.scala

```diff
@@ -91,6 +91,25 @@ private[tensor] class DenseTensor[@specialized(Float, Double) T: ClassTag](
     this
   }
 
+  override def cast[D: ClassTag](castTensor: Tensor[D])
+    (implicit ev1: TensorNumeric[D]): Tensor[T] = {
```

Comment:
the return type looks not right, should it be Tensor[D]?

Reply:
castTensor.applyFun[T](this, x => ev1.fromType[T](x)) ?

Reply:
`castTensor.applyFun[T](this, x => ev1.fromTypeT) ?` cannot work, because if you want to use fromType or toType, the T must be concrete type rather than a generic type.